src/base/path.cpp

```diff
@@ -45,8 +47,12 @@ const Qt::CaseSensitivity CASE_SENSITIVITY = Qt::CaseSensitive;
 const int PATHLIST_TYPEID = qRegisterMetaType<PathList>("PathList");
 
 Path::Path(const QString &pathStr)
-    : m_pathStr {QDir::cleanPath(pathStr)}
 {
+    const bool hasSeparator = std::any_of(pathStr.cbegin(), pathStr.cend(), [](const QChar c)
+    {
+        return (c == u'/') || (c == u'\\');
+    });
+    m_pathStr = hasSeparator ? QDir::cleanPath(pathStr) : pathStr;
```

Comment:
Maybe extract this code to some local `cleanPath()` to keep initializing `m_pathStr` in initialization section?

Reply:
Done.